api/app.py

```
@@ -137,30 +186,117 @@ def health():
+            return out
+        except HTTPException:
+            raise
+        except Exception as e:
+            logging.exception("Ingest failed")
+            run.add_outputs({"error": str(e)})
+            raise HTTPException(status_code=500, detail=str(e))
+
+# -------- PDF ingest with page-splitting + chunking --------
+@traceable
+@app.post("/ingest_pdf")
+async def ingest_pdf(file: UploadFile = File(...)):
```

Ruff (0.12.2)

Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)